server/sandbox_run_linux.go

```diff
@@ -375,53 +383,57 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 		}
 	}()
 
-	if _, err := s.ReservePodName(sbox.ID(), sbox.Name()); err != nil {
-		reservedID, getErr := s.PodIDForName(sbox.Name())
+	if _, err := s.ReservePodName(sboxID, sboxName); err != nil {
+		reservedID, getErr := s.PodIDForName(sboxName)
 		if getErr != nil {
-			return nil, fmt.Errorf("failed to get ID of pod with reserved name (%s), after failing to reserve name with %w: %w", sbox.Name(), getErr, getErr)
+			return nil, fmt.Errorf("failed to get ID of pod with reserved name (%s), after failing to reserve name with %w: %w", sboxName, getErr, getErr)
 		}
 		// if we're able to find the sandbox, and it's created, this is actually a duplicate request
 		// Just return that sandbox
 		if reservedSbox := s.GetSandbox(reservedID); reservedSbox != nil && reservedSbox.Created() {
 			return &types.RunPodSandboxResponse{PodSandboxId: reservedID}, nil
 		}
-		cachedID, resourceErr := s.getResourceOrWait(ctx, sbox.Name(), "sandbox")
+		cachedID, resourceErr := s.getResourceOrWait(ctx, sboxName, "sandbox")
 		if resourceErr == nil {
 			return &types.RunPodSandboxResponse{PodSandboxId: cachedID}, nil
 		}
 		return nil, fmt.Errorf("%w: %w", resourceErr, err)
 	}
-	resourceCleaner.Add(ctx, "runSandbox: releasing pod sandbox name: "+sbox.Name(), func() error {
-		s.ReleasePodName(sbox.Name())
+	resourceCleaner.Add(ctx, "runSandbox: releasing pod sandbox name: "+sboxName, func() error {
+		s.ReleasePodName(sboxName)
 		return nil
 	})
 
-	s.resourceStore.SetStageForResource(ctx, sbox.Name(), "sandbox creating")
+	// TODO: Pass interface instead of individual field.
```

Comment:
what's your plan for TODO's? address as follow ups? if so, should we open an issue to track them?

Reply:
@xw19 Any inputs on this?
Could you also squash the commits into a single one and prepare you PR for merging?

Reply:
@haircommander @sohankunkerkar I have created two issues and we will work on them
https://github.com/cri-o/cri-o/issues/8823
https://github.com/cri-o/cri-o/issues/8822

Reply:
@sohankunkerkar Squashed into a single commit.